nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -542,6 +542,59 @@ contract StakingEscrow is Issuer {
         }
     }
 
+    /**
+    * @notice Batch deposit. Allowed only initial deposit for each staker
+    * @param _stakers Stakers
+    * @param _values Amount of tokens to deposit for each staker
+    * @param _periods Amount of periods during which tokens will be locked for each staker
+    */
+    function batchDeposit(
+        address[] calldata _stakers,
+        uint256[] calldata _numberOfSubStakes,
+        uint256[] calldata _values,
+        uint16[] calldata _periods
+    )
+        external
+    {
+        uint256 subStakesLength = _values.length;
+        require(_stakers.length != 0 &&
+            _stakers.length == _numberOfSubStakes.length &&
+            subStakesLength >= _stakers.length &&
+            _periods.length == subStakesLength);
+        uint16 previousPeriod = getCurrentPeriod() - 1;
+        uint16 nextPeriod = previousPeriod + 2;
+        uint256 sumValue = 0;
+
+        uint256 j = 0;
+        for (uint256 i = 0; i < _stakers.length; i++) {
+            address staker = _stakers[i];
+            uint256 numberOfSubStakes = _numberOfSubStakes[i];
+            require(numberOfSubStakes > 0 && subStakesLength >= i + numberOfSubStakes);
+            StakerInfo storage info = stakerInfo[staker];
+            require(info.subStakes.length == 0);
+            require(workerToStaker[staker] == address(0) || workerToStaker[staker] == info.worker,
+                "A staker can't be a worker for another staker");
+            stakers.push(staker);
```

Comment:
What if the staker already exists? Are we perhaps duplicating the staker in the `stakers` list? This part `workerToStaker[staker] == info.worker` in the last `require` made me think it's plausible that the staker already exists.

Reply:
Agreed, `workerToStaker[staker] == info.worker` is unreachable, removed